core/reflect.go

```diff
@@ -171,36 +173,45 @@ func parseContractMethods(in BaseContractInterface) (ContractMethods, error) {
 			continue
 		}
 
+		var (
+			methodName          = method.Name
+			methodNameTruncated string
+		)
+
 		switch {
-		case len(method.Name) > 4 && method.Name[0:4] == noBatchPrefix:
+		case len(methodName) > 4 && methodName[0:4] == noBatchPrefix:
 			nb = true
-			method.Name = method.Name[4:]
-		case len(method.Name) > 5 && method.Name[0:5] == queryPrefix:
+			methodNameTruncated = methodName[4:]
+		case len(methodName) > 5 && methodName[0:5] == queryPrefix:
 			query = true
 			nb = true
-			method.Name = method.Name[5:]
-		case len(method.Name) > 2 && method.Name[0:2] == txPrefix:
-			method.Name = method.Name[2:]
+			methodNameTruncated = methodName[5:]
+		case len(methodName) > 2 && methodName[0:2] == txPrefix:
+			methodNameTruncated = methodName[2:]
```

Comment:
Maybe use the HasPrefix method and redo the prefix cutoff?

Reply:
Good point.

Reply:
Good point.